src/elementor/infrastructure/request-post.php

```diff
@@ -0,0 +1,136 @@
+					case 'wpseo_elementor_save':
+						// phpcs:ignore WordPress.Security.NonceVerification.Missing -- Reason: We are not processing form information.
+						if ( isset( $_POST['post_id'] ) && \is_numeric( $_POST['post_id'] ) ) {
+							// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized,WordPress.Security.NonceVerification.Missing -- Reason: No sanitization needed because we cast to an integer,We are not processing form information.
+							return (int) \wp_unslash( $_POST['post_id'] );
+						}
+
+						break;
+					// Elementor editor AJAX request.
+					case 'elementor_ajax':
+						return $this->get_document_id();
+				}
+
+				break;
+		}
+
+		return null;
+	}
+
+	/**
+	 * Returns the server request method.
+	 *
+	 * @return string The server request method, in upper case.
+	 */
+	private function get_server_request_method(): ?string {
+		if ( ! isset( $_SERVER['REQUEST_METHOD'] ) ) {
+			return null;
+		}
+
+		if ( ! \is_string( $_SERVER['REQUEST_METHOD'] ) ) {
+			return null;
+		}
+
+		// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized -- Reason: We are only comparing it later.
+		return \strtoupper( \wp_unslash( $_SERVER['REQUEST_METHOD'] ) );
+	}
+
+	/**
+	 * Retrieves the action from the POST request.
+	 *
+	 * @return string|null The action or null if not found.
+	 */
+	private function get_post_action(): ?string {
+		// phpcs:ignore WordPress.Security.NonceVerification.Missing -- Reason: We are not processing form information.
+		if ( isset( $_POST['action'] ) && \is_string( $_POST['action'] ) ) {
+			// phpcs:ignore WordPress.Security.NonceVerification.Missing,WordPress.Security.ValidatedSanitizedInput.InputNotSanitized -- Reason: We are not processing form information, we are only strictly comparing.
+			return (string) \wp_unslash( $_POST['action'] );
+		}
+
+		return null;
+	}
+
+	/**
+	 * Retrieves the document ID from the POST request.
+	 *
+	 * Note: this is specific to Elementor' `elementor_ajax` action. And then the `get_document_config` internal action.
+	 * Currently, you can see this in play when:
+	 * - showing the Site Settings in the Elementor editor
+	 * - going to another Recent post/page in the Elementor editor V2
+	 *
+	 * @return int|null The document ID or null if not found.
+	 */
+	private function get_document_id(): ?int {
+		// phpcs:ignore WordPress.Security.NonceVerification.Missing -- Reason: We are not processing form information.
+		if ( ! ( isset( $_POST['actions'] ) && \is_string( $_POST['actions'] ) ) ) {
+			return null;
+		}
+
+		// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized,WordPress.Security.NonceVerification.Missing -- Reason: No sanitization needed because we cast to an integer (after JSON decode and type/exist checks),We are not processing form information.
+		$actions = \json_decode( \wp_unslash( $_POST['actions'] ), true );
+		if ( ! \is_array( $actions ) ) {
+			return null;
+		}
+
+		\reset( $actions );
```

Comment:
Why are these two needed? If this array contains 1 element with a random key you could use `https://www.php.net/manual/en/function.array-pop.php` and not need a $key value.

Reply:
Originally, I went for `array_first_key` here, but that is not there in PHP 7.2 😅 
I'll change it to `array_shift`. While currently indeed just 1 element, it feels safer? 🤷